src/csharp/Microsoft.Spark/Interop/Ipc/JvmBridge.cs

```diff
@@ -31,6 +31,8 @@ internal sealed class JvmBridge : IJvmBridge
         [ThreadStatic]
         private static MemoryStream s_payloadMemoryStream;
 
+        private readonly SemaphoreSlim _socketSemaphore =
+            new SemaphoreSlim(SparkEnvironment.ConfigurationService.GetBackendThreads());
```

Comment:
This should be less than `DOTNET_NUM_BACKEND_THREADS`, no? I still see the tests hanging.

Reply:
It was working fine leaving it as the same number in my local test.  I can try something less than though.

Reply:
Should we do a flat number less than the setting ?  ie, -3 ?  Also should we make sure that this setting has a minimum value set?  Like >= 10 ?  If no minimum value, then what should we do in cases that the value is set to something low, like 1-5 ?

Reply:
We could do -3 and set the minimum as 5, and let's log the final number.